In [12]:
import pandas as pd

# Load both datasets using relative paths
goodreads_df = pd.read_csv("../data/goodreads_books.csv")
ratings_df = pd.read_csv("../data/user_rating_0_to_1000.csv")

# Preview both
print("Goodreads Dataset:")
print(goodreads_df.head())

print("\nUser Ratings Dataset:")
print(ratings_df.head())


Goodreads Dataset:
   bookID                                              title  \
0       1  Harry Potter and the Half-Blood Prince (Harry ...   
1       2  Harry Potter and the Order of the Phoenix (Har...   
2       4  Harry Potter and the Chamber of Secrets (Harry...   
3       5  Harry Potter and the Prisoner of Azkaban (Harr...   
4       8  Harry Potter Boxed Set  Books 1-5 (Harry Potte...   

                      authors average_rating        isbn       isbn13  \
0  J.K. Rowling/Mary GrandPré           4.57   439785960  9.78044E+12   
1  J.K. Rowling/Mary GrandPré           4.49   439358078  9.78044E+12   
2                J.K. Rowling           4.42   439554896  9.78044E+12   
3  J.K. Rowling/Mary GrandPré           4.56  043965548X  9.78044E+12   
4  J.K. Rowling/Mary GrandPré           4.78   439682584  9.78044E+12   

  language_code   num_pages  ratings_count  text_reviews_count  \
0           eng         652        2095690               27591   
1           eng         8

In [13]:
# Dataset shapes
print("Goodreads shape:", goodreads_df.shape)
print("Ratings shape:", ratings_df.shape)

# Column info
print("\nGoodreads Columns:")
print(goodreads_df.columns)

print("\nRatings Columns:")
print(ratings_df.columns)

# Missing values
print("\nMissing values (Goodreads):")
print(goodreads_df.isnull().sum())

print("\nMissing values (Ratings):")
print(ratings_df.isnull().sum())


Goodreads shape: (11121, 13)
Ratings shape: (51945, 3)

Goodreads Columns:
Index(['bookID', 'title', 'authors', 'average_rating', 'isbn', 'isbn13',
       'language_code', '  num_pages', 'ratings_count', 'text_reviews_count',
       'publication_date', 'publisher', 'Unnamed: 12'],
      dtype='object')

Ratings Columns:
Index(['ID', 'Name', 'Rating'], dtype='object')

Missing values (Goodreads):
bookID                    0
title                     0
authors                   0
average_rating            0
isbn                      0
isbn13                    0
language_code             0
  num_pages               0
ratings_count             0
text_reviews_count        0
publication_date          0
publisher                 0
Unnamed: 12           11118
dtype: int64

Missing values (Ratings):
ID        0
Name      0
Rating    0
dtype: int64


###  Overview from EDA 

The Kaggle Goodreads dataset has 11,121 rows and 13 columns. The ratings dataset is larger, with 51,945 rows and only 3 columns.
I checked the column names in both datasets to see what kinds of data are available. The Goodreads data includes book titles, authors, average ratings, number of pages, review counts, and more. The ratings dataset looks like user-submitted file with ID, name, and a rating.
I looked for any missing values using .isnull() and .sum() and every column has a value in every row. 
The Goodreads dataset includes a column called "Unnamed: 12", which I need to remove and "  num_pages" column has extra spaces in its name. 



In [14]:
ratings_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51945 entries, 0 to 51944
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   ID      51945 non-null  int64 
 1   Name    51945 non-null  object
 2   Rating  51945 non-null  object
dtypes: int64(1), object(2)
memory usage: 1.2+ MB


In [15]:
print(ratings_df['Rating'].unique())


['it was amazing' 'really liked it' 'liked it' 'did not like it'
 'it was ok' "This user doesn't have any rating"]
